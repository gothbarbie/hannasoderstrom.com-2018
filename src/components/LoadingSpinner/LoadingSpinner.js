import React from 'react'
import styled, { keyframes } from 'styled-components'

import Icon from '../Icon'

const rotating = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`

const Spinner = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.15);
  display: flex;
  width: 100%;
  height: 100%;
  align-items: center;
  justify-content: center;

  & span {
    animation: ${rotating} 1s linear infinite;
  }
`

const LoadingSpinner = () => (
  <Spinner>
    <Icon name="spinner" />
  </Spinner>
)

export default LoadingSpinner
