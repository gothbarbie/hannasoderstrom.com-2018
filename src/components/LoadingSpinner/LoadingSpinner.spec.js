import React from 'react'
import { shallow } from 'enzyme'

import LoadingSpinner from './index'

describe('components/LoadingSpinner', () => {
  let component

  beforeEach(() => {
    component = shallow(<LoadingSpinner />)
  })

  it('renders', () => {
    expect(component).toMatchSnapshot()
  })
})
