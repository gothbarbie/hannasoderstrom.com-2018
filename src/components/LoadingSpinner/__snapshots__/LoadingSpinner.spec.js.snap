// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LoadingSpinner renders 1`] = `
<styled.div>
  <Icon
    name="spinner"
  />
</styled.div>
`;
